src/pyodide/types/FS.d.ts

```diff
@@ -1,18 +1,96 @@
+
+interface Reader {
+  read: (offset: number, dest: Uint8Array) => number;
+}
+
 interface FSInfo {
   children: Map<string, FSInfo> | undefined;
   mode: number;
-  type: number;
+  type: string;
   modtime: number;
   size: number;
   path: string;
   name: string;
   parts: string[];
   contentsOffset?: number;
+  reader?: Reader;
 }
 
+declare type MetadataFSInfo = Map<string, MetadataFSInfo>;
+
 interface FS {
   mkdir: (dirname: string) => void;
   mkdirTree: (dirname: string) => void;
   writeFile: (fname: string, contents: Uint8Array, options: object) => void;
   mount: (fs: object, options: { info?: FSInfo }, path: string) => void;
+  createNode: (parent: FSNode | null, name: string, mode: number) => FSNode;
+  isFile: (mode: number) => boolean;
+  genericErrors: Error[];
+}
+
+interface FSOps {
+  getNodeMode: (parent: FSNode | null, name: string, info: any) => {
+    permissions: number;
+    isDir: boolean;
+  };
+  setNodeAttributes: (node: FSNode, info: any, isDir: boolean) => void;
```

Comment:
Not sure if this is right, but would be good to change 

```suggestion
  getNodeMode: (parent: FSNode | null, name: string, info: MetadataFSInfo) => {
    permissions: number;
    isDir: boolean;
  };
  setNodeAttributes: (node: FSNode, info: MetadataFSInfo, isDir: boolean) => void;
```

Reply:
Technically this object can be any type, Emscripten just passes them around. We use Map, FSInfo, and number